### ISSUE TITLE
IDE0052 should not be reported for fields with method invocations

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#31582]
**Version Used**: VS 16.0 Preview 1

**Steps to Reproduce**:

1. Copy the code below into a C# 8 project with nullable reference types enabled.

```c#
public class Repro
{
    private object? o;

    public void Init()
    {
        o = new object();
    }

    public int M() => o!.GetHashCode();
}
```

**Expected Behavior**: IDE0052 is not reported for `o` because one of its member function is used in `M()`.

**Actual Behavior**: IDE0052 is reported for `o`.
